elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -48,109 +54,224 @@ defmodule Web.Settings.DNS do
+                          @doh_disabled && "opacity-50 cursor-not-allowed",
+                          "relative group"
+                        ]}
+                        title={@doh_disabled && "Coming soon"}
+                      >
+                        <div class="block">
+                          <div class="w-full font-semibold mb-3">
+                            <.icon name="hero-lock-closed" class="w-5 h-5 mr-1" /> Secure DNS
+                          </div>
+                          <div class="w-full text-sm">
+                            Use DNS-over-HTTPS from trusted providers.
+                          </div>
+                        </div>
+                      </label>
+                    </li>
+                    <li>
+                      <.input
+                        id="dns-type--custom"
+                        type="radio_button_group"
+                        field={dns_form[:type]}
+                        value="custom"
+                        checked={"#{dns_form[:type].value}" == "custom"}
+                        required
+                      />
+                      <label for="dns-type--custom" class={~w[
+                        inline-flex items-center justify-between w-full
+                        p-5 text-gray-500 bg-white border border-gray-200
+                        rounded cursor-pointer peer-checked:border-accent-500
+                        peer-checked:text-accent-500 hover:text-gray-600 hover:bg-gray-100
+                      ]}>
+                        <div class="block">
+                          <div class="w-full font-semibold mb-3">
+                            <.icon name="hero-cog-6-tooth" class="w-5 h-5 mr-1" /> Custom DNS
+                          </div>
+                          <div class="w-full text-sm">
+                            Configure your own DNS server addresses.
+                          </div>
+                        </div>
+                      </label>
+                    </li>
+                  </ul>
+                </div>
+
+                <div :if={"#{dns_form[:type].value}" == "secure"} class="mb-6">
+                  <label class="block mb-2 text-sm font-medium text-neutral-900">
+                    DNS-over-HTTPS Provider
+                  </label>
+                  <.input
+                    type="select"
+                    field={dns_form[:doh_provider]}
+                    options={[
+                      {"Google Public DNS", :google},
+                      {"Cloudflare DNS", :cloudflare},
+                      {"Quad9 DNS", :quad9},
+                      {"OpenDNS", :opendns}
+                    ]}
+                  />
+                  <p class="mt-4 text-sm text-neutral-500">
+                    <strong>Note:</strong>
+                    Secure DNS is only supported on very recent Clients. Ensure your users are using the latest version to benefit from Secure DNS.
```

Comment:
I decided to not bother with listening the specific client versions. It looked a bit odd having all these different versions there. I think just saying "Make sure they are using the latest one" is good enough.